### PR TITLE
[image-picker] Rework the `UIImagePickerPresentationStyle` enum

### PR DESCRIPTION
**Pull request by barthap**
# Why

Addresses https://github.com/expo/expo/pull/16919#discussion_r842591950 - the same has been done in that PR for `expo-web-browser`.

To address our `Expo Modules Style Guide`:
> enum-like parameters should be typed as TypeScript enums with UPPER_CASE values and these should store string values (strings looks far better in logs than numbers). Then, on native layer these string values should be mapped to the respective APIs parameters → use Sweet API auto-conversion for that.

# How

- Deleted the `UIImagePickerPresentationStyle.BlurOverFullScreen` option. It wasn't working anyway because it was `tvOS` only.
- Deprecated all `PascalCase` values of the `UIImagePickerPresentationStyle` enum in favor of their `SNAKE_UPPERCASE` counterparts.
- Underlying values of the `UIImagePickerPresentationStyle` are now strings. They were integers before.

# Test Plan

NCL

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-image-picker/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,12 @@
 
 - Fix failure on Android when `allowsEditing` is `true` and non-jpeg file picked. ([#16615](https://github.com/expo/expo/pull/16615) by [@mnightingale](https://github.com/mnightingale))
 
+### ⚠ Notices
+
+- Deleted the `UIImagePickerPresentationStyle.BlurOverFullScreen` option.
```
[the remark this answers is not in the document]
Comment (bbarthec) [reply packages/expo-image-picker/CHANGELOG.md:19]:
```suggestion
- Deleted the `UIImagePickerPresentationStyle.BlurOverFullScreen` option as it does not work on iOS. ([#16925](https://github.com/expo/expo/pull/16925) by [@barthap](https://github.com/barthap))
```

File: packages/expo-image-picker/src/ImagePicker.types.ts [hunk 2]
```diff
@@ -140,15 +140,73 @@ export enum UIImagePickerControllerQualityType {
...
-  PageSheet = 1,
-  FormSheet = 2,
-  CurrentContext = 3,
-  OverFullScreen = 5,
-  OverCurrentContext = 6,
-  Popover = 7,
-  BlurOverFullScreen = 8,
-  Automatic = -2,
+  /**
+   * A presentation style in which the presented picker covers the screen.
```
Comment (bbarthec) [inline packages/expo-image-picker/src/ImagePicker.types.ts:144]:
```suggestion
   * A presentation style in which the presented picker covers the screen.
   * @see https://developer.apple.com/documentation/uikit/uimodalpresentationstyle/uimodalpresentationfullscreen
```

File: packages/expo-image-picker/src/ImagePicker.types.ts [hunk 3]
```diff
@@ -140,15 +140,73 @@ export enum UIImagePickerControllerQualityType {
...
+   */
+  FULL_SCREEN = 'fullScreen',
+  /**
+   * A presentation style that partially covers the underlying content.
```
Comment (bbarthec) [inline packages/expo-image-picker/src/ImagePicker.types.ts:148]:
```suggestion
   * A presentation style that partially covers the underlying content.
   * @see https://developer.apple.com/documentation/uikit/uimodalpresentationstyle/uimodalpresentationpagesheet
```

**Review by tsapeta — APPROVED** [review]

**Review by bbarthec — APPROVED** [review]
Looking good 👍 
What would you say about adding reference to the apple docs for every `UIImagePickerPresentationStyle` property? 🤔

**Review by lukmccall — APPROVED** [review]

**Pull request merged**